### PR TITLE
2 implement ili9163c driver in 6tron workspace

### LINKED ISSUE #2: Implement ili9163c driver in 6tron-workspace

### PR DESCRIPTION
**Pull request by andiaye33**
This issue perform the following tasks:

 **For ili9163c.c:**
- Configured MIPI-DBI interface for communication.
- Implement general functions : initialization, memory area, pixel format, and orientation setting.
- Added display blanking on/off function.
- Add hardware reset functionality.
- Add Iregister initialization.

**For ili9163c.h**
- Add macros, register and structures of the ILI9163C.

**For Samples** 
- Update samples to demonstrate the usage of the ili9163c driver function.
- Use LVGL and PWM lib.

### CODE REVIEW

File: drivers/display/ilitek/ili9163c/Kconfig [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
-config ILI9163C
-    bool "ILI9163C Display Driver"
+menuconfig ILI9163C
+    bool "ILI9163C display driver"
     default y
     depends on DT_HAS_ILITEK_ILI9163C_ENABLED
-    select ILI9XXX
-    select SPI
+    select MIPI_DBI
+    select MIPI_DBI_SPI
```
Comment (clementlgl) [inline drivers/display/ilitek/ili9163c/Kconfig]: I think we need to select only `MIPI_DBI`

File: samples/CMakeLists.txt [hunk 2]
```diff
@@ -1,9 +1,12 @@
-# Copyright (c) 2024 CATIE
+# Copyright (c) 2024, CATIE
 # SPDX-License-Identifier: Apache-2.0
 
 cmake_minimum_required(VERSION 3.20.0)
+
+set(SHIELD zest_display_lcd)
+
```
Comment (clementlgl) [inline samples/CMakeLists.txt]:
```suggestion
```

File: samples/CMakeLists.txt [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 project(ili9163c)
 
-FILE(GLOB app_sources src/main.c)
+FILE(GLOB app_sources src/*.c)
```
Comment (clementlgl) [inline samples/CMakeLists.txt]: There is only one `.c` file 
```suggestion
FILE(GLOB app_sources src/main.c)
```

**Comment by Sokario** [comment]
@andiaye33: The ```app.overlay``` is missing from the repository.

File: samples/prj.conf [hunk 4]
```diff
@@ -1,2 +1,25 @@
-CONFIG_STDOUT_CONSOLE=y
+CONFIG_LV_Z_MEM_POOL_SIZE=16384
+CONFIG_LV_Z_SHELL=y
+CONFIG_MAIN_STACK_SIZE=2048
```
Comment (clementlgl) [inline samples/prj.conf:2]: Why change the main stack size ?

File: samples/prj.conf [hunk 5]
```diff
@@ -1,2 +1,25 @@
...
 CONFIG_DISPLAY=y
+CONFIG_DISPLAY_LOG_LEVEL_ERR=y
+
+CONFIG_GPIO=y
+
+
+CONFIG_LOG_DEFAULT_LEVEL=4
+
+CONFIG_PWM=y
+CONFIG_LOG=y
```
Comment (clementlgl) [inline samples/prj.conf]: I think the maximum log level for example should be Info (3) or Default

File: samples/prj.conf [hunk 6]
```diff
@@ -1,2 +1,25 @@
...
+CONFIG_SHELL=y
```
Comment (clementlgl) [inline samples/prj.conf]: Shell is used ?

File: samples/src/main.c [hunk 7]
```diff
@@ -3,27 +3,197 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include <zephyr/kernel.h>
 #include <zephyr/device.h>
-#include <zephyr/drivers/sensor.h>
+#include <zephyr/devicetree.h>
```
Comment (clementlgl) [inline samples/src/main.c]:
```suggestion
```

Already included in `device.h`

File: samples/src/main.c [hunk 8]
```diff
@@ -3,27 +3,197 @@
...
+#include <zephyr/drivers/display.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pwm.h>
+#include <lvgl.h>
+#include <stdio.h>
+#include <string.h>
+#include <zephyr/kernel.h>
+#include <lvgl_input_device.h>
```
Comment (clementlgl) [inline samples/src/main.c]: On this example we don't have any input device. I think we didn't need to include `lvgl_input_device.h`

File: samples/src/main.c [hunk 9]
```diff
@@ -3,27 +3,197 @@
...
+	}
+
+	lv_label_set_text(hello_world_label, "6TRON BY CATIE!");
+	lv_obj_align(hello_world_label, LV_ALIGN_CENTER, 0, 0);
+
+	count_label = lv_label_create(lv_scr_act());
+	lv_obj_align(count_label, LV_ALIGN_BOTTOM_MID, 0, 0);
+
+	lv_task_handler();
+	display_blanking_off(display_dev);
```
Comment (clementlgl) [inline samples/src/main.c]: Rework to use only the ilitek display, without lvgl_input

**Review by clementlgl — CHANGES_REQUESTED** [review]

File: samples/prj.conf (hunk 5 above)
Comment (andiaye33) [reply samples/prj.conf]: Will be done in next commit

File: samples/prj.conf (hunk 6 above)
Comment (andiaye33) [reply samples/prj.conf]: No, it will removed in next commit

File: samples/src/main.c (hunk 8 above)
Comment (andiaye33) [reply samples/src/main.c]: Indeed we don't need input device, all related code will be deleted in next commit

File: samples/src/main.c (hunk 9 above)
Comment (andiaye33) [reply samples/src/main.c]: Resolved in [1107a99](https://github.com/catie-aq/zephyr_ilitek-ili9163c/pull/3/commits/1107a9950e66be1f2fa57e347fc6bf2b0cb3b9f8).

File: samples/src/main.c (hunk 8 above)
Comment (andiaye33) [reply samples/src/main.c]: Resolved in [1107a99](https://github.com/catie-aq/zephyr_ilitek-ili9163c/pull/3/commits/1107a9950e66be1f2fa57e347fc6bf2b0cb3b9f8).

File: samples/prj.conf (hunk 6 above)
Comment (andiaye33) [reply samples/prj.conf]: Resolved in [0864c97](https://github.com/catie-aq/zephyr_ilitek-ili9163c/pull/3/commits/0864c971048ea99dbde05a4a1ab9568a47fd966e).

File: samples/prj.conf (hunk 5 above)
Comment (andiaye33) [reply samples/prj.conf]: Resolved in [0864c97](https://github.com/catie-aq/zephyr_ilitek-ili9163c/pull/3/commits/0864c971048ea99dbde05a4a1ab9568a47fd966e).

File: samples/CMakeLists.txt (hunk 3 above)
Comment (andiaye33) [reply samples/CMakeLists.txt]: Resolved in [9d3e388](https://github.com/catie-aq/zephyr_ilitek-ili9163c/pull/3/commits/9d3e388f593571c4797e21606e0134fa4e9482c9).

File: samples/CMakeLists.txt (hunk 2 above)
Comment (andiaye33) [reply samples/CMakeLists.txt]: Resolved in [cd92ff1](https://github.com/catie-aq/zephyr_ilitek-ili9163c/pull/3/commits/cd92ff1a017f5defaaaeda6a3688e882728c5d1b).

File: drivers/display/ilitek/ili9163c/Kconfig (hunk 1 above)
Comment (andiaye33) [reply drivers/display/ilitek/ili9163c/Kconfig]: Resolved in [c86a28e](https://github.com/catie-aq/zephyr_ilitek-ili9163c/pull/3/commits/c86a28e946dccfc85693e460b073e29c4d36e9d8).

File: samples/prj.conf (hunk 4 above)
Comment (andiaye33) [reply samples/prj.conf:2]: Zephyr main default  stack is unsuficiant for  this project du to lvgl

File: samples/app.overlay [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+			 pwctrl1 = [0A 02];
+			 pwctrl2 = [02];
+			 vmctrl1 = [55 60];
+			 vmctrl2 = [45];
+			 gamset = [04];  
+			 gamadj = [01];    
+			 dispinv = [07];    
+		 };
+	 };
+ };
```
Comment (clementlgl) [inline samples/app.overlay]:
```suggestion
 };

```

**Review by clementlgl — CHANGES_REQUESTED** [review]

File: samples/app.overlay [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+ 
+	 mipi_dbi {
+		 compatible = "zephyr,mipi-dbi-spi";
+		 dc-gpios = <&sixtron_connector DIO2 GPIO_ACTIVE_HIGH>;  /* Data/Command */
+		 spi-dev = <&sixtron_spi>;
+		 #address-cells = <1>;
+		 #size-cells = <0>;
+		 write-only;
+ 
+		 ili9163c0: ili9163c@0 {
```
Comment (clementlgl) [inline samples/app.overlay]: I think de label should be `ili9163c` (without the `0` at the end).

File: samples/app.overlay [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+
+ #include <zephyr/dt-bindings/input/input-event-codes.h>
+ #include <zephyr/dt-bindings/lvgl/lvgl.h>
+ #include <zephyr/dt-bindings/gpio/sixtron-header.h>
+ #include <zephyr/dt-bindings/display/ili9xxx.h>
+ 
+  &timers1 {
+	 status = "okay";
+	 st,prescaler = <1000>;
+ };
```
Comment (clementlgl) [inline samples/app.overlay]: For better readability, the timer should be configured after the aliases, and other "new device" declarations.

File: samples/app.overlay [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+			 frmctr1 = [08 08];
+			 pgamctrl = [3F 25 1C 1E 20 12 2A 90 24 11 00 00 00 00 00];
+			 ngamctrl = [20 20 20 20 05 00 15 A7 3D 18 25 2A 2B 2B 3A];
+			 pwctrl1 = [0A 02];
+			 pwctrl2 = [02];
+			 vmctrl1 = [55 60];
+			 vmctrl2 = [45];
+			 gamset = [04];  
+			 gamadj = [01];    
+			 dispinv = [07];
```
Comment (clementlgl) [inline samples/app.overlay]: I couldn't find these values when I developed the Mbed OS driver, where did you find them?

**Review by clementlgl — CHANGES_REQUESTED** [review]
Just a comment about the device tree format

File: samples/app.overlay (hunk 13 above)
Comment (Sokario) [reply samples/app.overlay]: I check with datasheet and made every config as default value

**Review by clementlgl — DISMISSED** [review]

File: README.md [hunk 14]
```diff
@@ -1,3 +1,21 @@
...
+- Zest_Display_LCD
+- Any Zest Core board
+
+## Usage
+This dispalay driver can be used to display and draw text, images, and shapes in highly readable form.
+
+### Build
+Use the Zest_Display_LCD shield:
+```bash
+west build -b <board> samples -- -DSHIELD=zest_display_lcd
```
Comment (Sokario) [inline README.md]: Is it correct to ask in the driver to use a shield?

File: samples/README.md [hunk 15]
```diff
@@ -0,0 +1,50 @@
+# Overview
+
+This sample application provides an example usage of the display ILI9163C from ILITEK.
+The example includes PWM control for backlight dimming and displays text and a counter on the screen.
+
+- Displays "6TRON BY CATIE!" on the screen. 
+- Shows an incrementing counter at the bottom of the screen.
```
Comment (Sokario) [inline samples/README.md]: Might be better to say it count pasts seconds.

File: samples/README.md [hunk 16]
```diff
@@ -0,0 +1,50 @@
+# Overview
+
+This sample application provides an example usage of the display ILI9163C from ILITEK.
+The example includes PWM control for backlight dimming and displays text and a counter on the screen.
+
+- Displays "6TRON BY CATIE!" on the screen. 
+- Shows an incrementing counter at the bottom of the screen. 
+- Uses PWM to control the backlight brightness in a pulsing pattern. 
+- Refreshes the display content periodically using LittlevGL (LVGL).
```
Comment (Sokario) [inline samples/README.md]: I think we already discussed this point but I think the sample shouldn't depends on LVGL. WE will have a example on how to use the display with LVGL.

File: samples/README.md [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+
+> [!NOTE]
+>  Ensure the display and PWM hardware configurations match the setup below for correct operation.
```
Comment (Sokario) [inline samples/README.md]: Might be good to add an example  for the pwm setup you have right now. It is not ubiquitus to everyone how to setup the pwm correctly for the driver.

File: samples/README.md [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+
+# Requirements
+
+- Hardware: Zest Display LCD shield with ILI9163C display and any compatible Zest Core board.
```
Comment (Sokario) [inline samples/README.md]: This driver only depends on ILI9163C. It is not a depot where the shield should be mentionned.

File: samples/README.md [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+# Building and Running
+
+```shell
+cd <driver_directory>
+west build -p always -b <BOARD> samples/
+west flash
+```
+
+# Sample Output
+
```
Comment (Sokario) [inline samples/README.md]: I don't know if it is a good practice but a screen of the display might be wanted here.

**Review by Sokario — CHANGES_REQUESTED** [review]

**Review by Sokario — COMMENTED** [review]
Good first README but some information might be a little of in my opinion.

**Comment by clementlgl** [comment]
I think the readme should not mention `Zest`, because the driver is generic and not specific to 6TRON.
We have writtent this information on some drivers, but maybe it's a mistake.

Currently the cookiecutter adds only a title and a simple description of the driver. Zephyr documents only examples ...
Maybe we need to document a little more what the driver can do.

File: samples/README.md (hunk 19 above)
Comment (andiaye33) [reply samples/README.md]: We must ask to @clementlgl

File: samples/README.md (hunk 18 above)
Comment (andiaye33) [reply samples/README.md]: Resolved in 1ea59ed680dc881d3afea2fa9cafe1a5cf060ca0.

File: samples/README.md (hunk 17 above)
Comment (andiaye33) [reply samples/README.md]: We must ask to @clementlgl

File: samples/README.md (hunk 16 above)
Comment (andiaye33) [reply samples/README.md]: Resolved in 1ea59ed680dc881d3afea2fa9cafe1a5cf060ca0.

File: samples/README.md (hunk 15 above)
Comment (andiaye33) [reply samples/README.md]: Resolved in 1ea59ed680dc881d3afea2fa9cafe1a5cf060ca0.

File: README.md (hunk 14 above)
Comment (andiaye33) [reply README.md]: Resolved in 1ea59ed680dc881d3afea2fa9cafe1a5cf060ca0.

File: samples/README.md [hunk 20]
```diff
@@ -0,0 +1,42 @@
+.. 6tron :code-sample:: display
+   :name: Display
+   :relevant-api: display_interface
+
+   Draw basic rectangles from zephyr display samples [https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/drivers/display].
```
Comment (clementlgl) [inline samples/README.md]: These lines should not exist...

File: samples/README.md [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+    - Bottom Left: Gray
+- Uses PWM to control the backlight brightness.
+
+> [!NOTE]
+> Ensure the display and PWM hardware configurations match the setup below for correct operation.
+
+# Requirements
+
+- **Hardware:**
+    - Any compatible Zest Core board.
```
Comment (clementlgl) [inline samples/README.md]: Comme je l'ai dit dans un commentaire précédent, il s'agit d'un conducteur, et nous n'avons pas besoin de parler de Zest.

File: samples/prj.conf [hunk 22]
```diff
@@ -1,2 +1,11 @@
...
+CONFIG_MAIN_STACK_SIZE=2048
+
 CONFIG_DISPLAY=y
+CONFIG_DISPLAY_LOG_LEVEL_ERR=y
+
+CONFIG_GPIO=y
+
+CONFIG_PWM=y
+CONFIG_LOG=y
+CONFIG_LOG_DEFAULT_LEVEL=3
```
Comment (clementlgl) [inline samples/prj.conf]:
```suggestion
CONFIG_LOG_DEFAULT_LEVEL=3

```

**Review by clementlgl — CHANGES_REQUESTED** [review]
Need a few changes to the Readme. The format doesn't match the cookiecutter. But maybe it needs to evolve too.

**Review by clementlgl — DISMISSED** [review]

File: README.md [hunk 23]
```diff
@@ -1,3 +1,16 @@
...
 
-ILITEK ILI9163C display controller sensor driver for Zephyr OS.
+ILITEK ILI9163C display driver for Zephyr OS.
+
+Based on [ILI9163C Datasheet](https://www.buydisplay.com/download/ic/ILI9163.pdf).
+Implemented feature:
+
+- [X] Display Initialization
+- [X] Pixel Format Configuration
+- [X] Orientation Adjustment.
```
Comment (Sokario) [inline README.md]:
```suggestion
- [] Orientation Adjustment.
```

File: samples/README.md [hunk 24]
```diff
@@ -0,0 +1,36 @@
...
+
+- Draws rectangles onto the display in clockwise using the CATIE color palette :
+    - Top Left: Orange
+    - Top Right: Blue
+    - Bottom Right: Green
+    - Bottom Left: Gray
+- Uses PWM to control the backlight brightness.
+
+> [!NOTE]
+> Ensure the display and PWM hardware configurations match the setup below for correct operation.
```
Comment (Sokario) [inline samples/README.md]:
```suggestion
> Ensure the display and PWM hardware configurations match the requirements below for correct operation.
```

File: samples/README.md [hunk 25]
```diff
@@ -0,0 +1,36 @@
...
+
+# Requirements
+
+- **Hardware:**
+    - Board supporting I2C communication.
+- **Configuration Options:**
+    - Set `CONFIG_DISPLAY=y` in `prj.conf` to use the DISPLAY API.
+    - Set `CONFIG_PWM=y` in `prj.conf` to use the PWM API.
```
Comment (Sokario) [inline samples/README.md]:
```suggestion
    - Set `CONFIG_PWM=y` in `prj.conf` to use the PWM API for backlight control.
```

**Review by Sokario — CHANGES_REQUESTED** [review]

**Review by Sokario — APPROVED** [review]

**Review by clementlgl — APPROVED** [review]

**Pull request merged**